pkg/resources/resource_helpers_read.go

```diff
@@ -49,3 +49,20 @@ func setBooleanStringFromBoolProperty(d *schema.ResourceData, key string, proper
 	}
 	return nil
 }
+
+func attributeMappedValueReadIfNotEmptyElse[T, R any](d *schema.ResourceData, key string, value *T, mapper func(*T) (R, error), defaultValue any) error {
```

Comment:
Maybe `attributeMappedValueReadIfNotEmptyElse` -> `attributeMappedValueReadOrDefualt`?

Reply:
Additionally, default value should be also type R, right?

Reply:
True, applied both